generators/git/index.js

```diff
@@ -51,10 +58,14 @@ module.exports = generators.Base.extend({
 
     this.pkg.repository = this.pkg.repository || repository;
 
-    this.fs.writeJSON('package.json', this.pkg);
+    this.fs.writeJSON(this.destinationPath(this.options.generateInto, 'package.json'), this.pkg);
   },
 
   end: function () {
+    if (this.options.generateInto) {
```

Comment:
Okay, I get it sorry.

So just define a different CWD to the spawnCommand. Don't change the destinationRoot.